### ISSUE TITLE
tf.map_fn on RaggedTensors crash during gradient computation on a GPU

### ISSUE THREAD

**Issue by foxik** [tensorflow/tensorflow#55475]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Colab**
- TensorFlow installed from (source or binary): **binary**
- TensorFlow version (use command below): **2.8**
- Python version: **3.7**

**Describe the current behavior**

When some loss (`tf.losses.SparseCategoricalCrossentropy`, `tf.losses.CategoricalCrossentropy`, `tf.losses.BinaryCrossentropy`, or `tf.losses.MeanSquaredError`) is used on Ragged tensors, which is computed via a `tf.map_fn` on a `RaggedTensor`, that the gradient computation on a GPU crashes with
```
Node: 'Adam/gradients/zeros_like_2'
2 root error(s) found.
  (0) INTERNAL:  No unary variant unary_op function found for op ZEROS_LIKE Variant type_name: RaggedTensorVariant for device type: GPU
	 [[{{node Adam/gradients/zeros_like_2}}]]
	 [[binary_crossentropy/map/while/loop_body_control/_124/_67]]
  (1) INTERNAL:  No unary variant unary_op function found for op ZEROS_LIKE Variant type_name: RaggedTensorVariant for device type: GPU
	 [[{{node Adam/gradients/zeros_like_2}}]]
0 successful operations.
0 derived errors ignored. [Op:__inference_train_function_16690]
```

The computation does not crash on a CPU and it does not crash when `tf.function`s are executed eagerly.

Also, if the `tf.map_fn` is circumvented by using the following argument to compile
```python
  loss=lambda yt, yp: tf.losses.BinaryCrossentropy()(yt. values, yp.values)
```
it works on GPU without a crash.

**Describe the expected behavior**

The code does not crash on a GPU.

- Do you want to contribute a PR? (yes/no): **no**

**Standalone code to reproduce the issue**

A simple Colab reproducing the error is here: https://colab.research.google.com/drive/1OELAhvpQHhaz3sOYabf4SdBqKlQCjNjs?usp=sharing

**Other info / logs**

The `map_fn` used is here: https://github.com/keras-team/keras/blob/2db5acf3e3c5904b014cb409d3c514bef44f9640/keras/losses.py#L1408

**Comment by foxik** [comment]
Note that I also opened an issue in the Keras repository https://github.com/keras-team/keras/issues/16354 , where we discuss whether we should avoid the `tf.map_fn` on the RaggedTensors, because it can probably be avoided -- the metrics with ragged tensors take a different approach, and instead of a ragged map, they use `flat_values`, see https://github.com/keras-team/keras/blob/2db5acf3e3c5904b014cb409d3c514bef44f9640/keras/utils/metrics_utils.py#L800 .

**Comment by sushreebarsa** [comment]
@chunduriv I was able to reproduce the issue on colab using TF v2.8.0 ,tf-nightly on both [gpu](https://colab.research.google.com/gist/sushreebarsa/014fdaf99d8e375ce618c3c5aee9d150/untitled612.ipynb#scrollTo=cDbvjfYIh2Nu) and [cpu](https://colab.research.google.com/gist/sushreebarsa/22c50fcc02dd280c38bdfffc5c3f2235/55475-cpu.ipynb) , please find the attached gists for reference.Thanks!

**Comment by foxik** [comment]
Oh, I was just pointed to me (by djoshea) that this is a duplicate of 46635, so closing.

**Comment by foxik** [comment]
Closing as a duplicate of #46635 .
